include/openthread/coap.h

```diff
@@ -153,20 +153,67 @@ typedef struct otCoapOption
 
 /**
  * CoAP Content Format codes.  The full list is documented at
- * https://tools.ietf.org/html/rfc7252#page-92
- *
+ * https://www.iana.org/assignments/core-parameters/core-parameters.xhtml#content-formats
  */
 typedef enum otCoapOptionContentFormat
 {
-    OT_COAP_OPTION_CONTENT_FORMAT_TEXT_PLAIN   = 0,  ///< text/plain
-    OT_COAP_OPTION_CONTENT_FORMAT_LINK_FORMAT  = 40, ///< application/link-format
-    OT_COAP_OPTION_CONTENT_FORMAT_XML          = 41, ///< application/xml
-    OT_COAP_OPTION_CONTENT_FORMAT_OCTET_STREAM = 42, ///< application/octet-stream
-    OT_COAP_OPTION_CONTENT_FORMAT_EXI          = 47, ///< application/exi
-    OT_COAP_OPTION_CONTENT_FORMAT_JSON         = 50, ///< application/json
-    OT_COAP_OPTION_CONTENT_FORMAT_PKCS10       = 70, ///< application/pkcs10
-    OT_COAP_OPTION_CONTENT_FORMAT_PKCS7        = 80, ///< application/pkcs7
-    OT_COAP_OPTION_CONTENT_FORMAT_JWS          = 101 ///< application/json-web-signature
+    /** text/plain; charset=utf-8: [RFC2046][RFC3676][RFC5147] */
```

Comment:
What about to make these enums more readable? I am thinking e.g. about formating similar to `otError `in error.h.

```
    /**
     * Media Type: text/plain; charset=utf-8 [RFC2046][RFC3676][RFC5147]
     */
    OT_COAP_OPTION_CONTENT_FORMAT_TEXT_PLAIN = 0,

    /**
     * Media Type: application/cose; cose-type="cose-encrypt0" [RFC8152]
     */
    OT_COAP_OPTION_CONTENT_FORMAT_COSE_ENCRYPT0 = 16,

    /**
     * Media Type: application/cose; cose-type="cose-mac0" [RFC8152]
     */
    OT_COAP_OPTION_CONTENT_FORMAT_COSE_MAC0 = 17,

    /**
     * Media Type: application/cose; cose-type="cose-sign1" [RFC8152]
     */
    OT_COAP_OPTION_CONTENT_FORMAT_COSE_SIGN1 = 18,

    [...]
```

Or sth like: `s/content-type/media-type` is also fine for me.

```
    /**
     * content-type=text/plain; charset=utf-8 [RFC2046][RFC3676][RFC5147]
     */
    OT_COAP_OPTION_CONTENT_FORMAT_TEXT_PLAIN = 0,

    /**
     * content-type=application/cose; cose-type="cose-encrypt0" [RFC8152]
     */
    OT_COAP_OPTION_CONTENT_FORMAT_COSE_ENCRYPT0 = 16,

    /**
     * content-type=application/cose; cose-type="cose-mac0" [RFC8152]
     */
    OT_COAP_OPTION_CONTENT_FORMAT_COSE_MAC0 = 17,

    /**
     * content-type=application/cose; cose-type="cose-sign1" [RFC8152]
     */
    OT_COAP_OPTION_CONTENT_FORMAT_COSE_SIGN1 = 18,

    [...]
```

Reply:
I did try aligning the values up, only to have `make pretty` revert those changes.  I can try again though.